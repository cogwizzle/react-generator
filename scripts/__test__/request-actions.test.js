const { generateRequestHookActions } = require('../request-hook-actions')

it('no semi, no jest', () => {
  const results = generateRequestHookActions({}, 'js')
  expect(results).toMatchSnapshot()
})

it('no semi, jest', () => {
  const results = generateRequestHookActions({ isJest: true }, 'js')
  expect(results).toMatchSnapshot()
})

it('semi, no jest', () => {
  const results = generateRequestHookActions({ isSemicolons: true }, 'js')
  expect(results).toMatchSnapshot()
})

it('semi, jest', () => {
  const results = generateRequestHookActions({
    isSemicolons: true,
    isJest: true,
  }, 'js')
  expect(results).toMatchSnapshot()
})

