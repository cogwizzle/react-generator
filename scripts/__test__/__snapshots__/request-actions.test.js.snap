// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`no semi, jest 1`] = `
Array [
  Object {
    "path": "/home/jfehrman/Development/react-generator/src/hooks/use-request-{{snakeCase name}}.js",
    "skipIfExists": false,
    "template": undefined,
    "type": "add",
  },
  Object {
    "path": "/home/jfehrman/Development/react-generator/src/hooks/use-request-{{snakeCase name}}.js",
    "pattern": /;\\\\n/g,
    "template": "
",
    "type": "modify",
  },
  Object {
    "path": "/home/jfehrman/Development/react-generator/src/hooks/__test__/use-request-{{snakeCase name}}.test.js",
    "skipIfExists": false,
    "template": undefined,
    "type": "add",
  },
  Object {
    "path": "/home/jfehrman/Development/react-generator/src/hooks/__test__/use-request-{{snakeCase name}}.test.js",
    "pattern": /;\\\\n/g,
    "template": "
",
    "type": "modify",
  },
]
`;

exports[`no semi, no jest 1`] = `
Array [
  Object {
    "path": "/home/jfehrman/Development/react-generator/src/hooks/use-request-{{snakeCase name}}.js",
    "skipIfExists": false,
    "template": undefined,
    "type": "add",
  },
  Object {
    "path": "/home/jfehrman/Development/react-generator/src/hooks/use-request-{{snakeCase name}}.js",
    "pattern": /;\\\\n/g,
    "template": "
",
    "type": "modify",
  },
]
`;

exports[`semi, jest 1`] = `
Array [
  Object {
    "path": "/home/jfehrman/Development/react-generator/src/hooks/use-request-{{snakeCase name}}.js",
    "skipIfExists": false,
    "template": undefined,
    "type": "add",
  },
  Object {
    "path": "/home/jfehrman/Development/react-generator/src/hooks/__test__/use-request-{{snakeCase name}}.test.js",
    "skipIfExists": false,
    "template": undefined,
    "type": "add",
  },
]
`;

exports[`semi, no jest 1`] = `
Array [
  Object {
    "path": "/home/jfehrman/Development/react-generator/src/hooks/use-request-{{snakeCase name}}.js",
    "skipIfExists": false,
    "template": undefined,
    "type": "add",
  },
]
`;
